applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -35,32 +35,44 @@ enum state {
 
 static enum state module_state = STATE_DISABLED;
 
+#define FOR_DVFS_STATE_LIST(MACRO)	\
+	MACRO(DVFS_STATE_INITIALIZING)	\
+	MACRO(DVFS_STATE_LLPM_CONNECTED)\
+	MACRO(DVFS_STATE_USB_CONNECTED)	\
+	MACRO(DVFS_STATE_CONFIG_CHANNEL)\
+	MACRO(DVFS_STATE_SMP_TRANSFER)	\
+	MACRO(DVFS_STATE_COUNT)
+
+#define DEFINE_ENUMERATION(name, ...) name,
+#define HIGH_FREQ_BIT(name, ...)	IF_ENABLED(CONFIG_ ## name ## _FREQ_HIGH,  BIT(name) |)
+#define MEDLOW_FREQ_BIT(name, ...)	IF_ENABLED(CONFIG_ ## name ## _FREQ_MEDLOW, BIT(name) |)
+#define INITIAL_ACTIVE_BIT(name, ...)	IF_ENABLED(CONFIG_ ## name ## _INITIAL_ACTIVE, BIT(name) |)
+#define GET_ENUM_NAME(name, ...)	case name: return #name;
+#define SET_STATE_TIMEOUT(name, ...)					\
+		IF_ENABLED(CONFIG_ ## name ## _ENABLE,			\
+		([name] = {						\
+			.timeout_ms = CONFIG_ ## name ## _TIMEOUT_MS,	\
+		},))
+
 enum dvfs_state {
-	DVFS_STATE_INITIALIZING,
-	DVFS_STATE_LLPM_CONNECTED,
-	DVFS_STATE_USB_CONNECTED,
-	DVFS_STATE_CONFIG_CHANNEL,
-	DVFS_STATE_SMP_TRANSFER,
-	DVFS_STATE_COUNT
+	FOR_DVFS_STATE_LIST(DEFINE_ENUMERATION)
 };
 
-static const uint8_t dvfs_high_freq_bitmask = BIT(DVFS_STATE_INITIALIZING) |
-					      BIT(DVFS_STATE_USB_CONNECTED) |
-					      BIT(DVFS_STATE_CONFIG_CHANNEL) |
-					      BIT(DVFS_STATE_SMP_TRANSFER);
-static const uint8_t dvfs_medlow_freq_bitmask = BIT(DVFS_STATE_LLPM_CONNECTED);
+static const uint8_t dvfs_high_freq_bitmask = FOR_DVFS_STATE_LIST(HIGH_FREQ_BIT) 0;
+static const uint8_t dvfs_medlow_freq_bitmask = FOR_DVFS_STATE_LIST(MEDLOW_FREQ_BIT) 0;
 
-/* Binary mask tracking which states are requested.
- * We start with DVFS_STATE_INITIALIZING on as it is active on start.
- */
-static uint8_t dfvs_requests_state_bitmask = BIT(DVFS_STATE_INITIALIZING);
+/* Binary mask tracking which states are requested. */
+static uint8_t dfvs_requests_state_bitmask = FOR_DVFS_STATE_LIST(INITIAL_ACTIVE_BIT) 0;
+
+static enum dvfs_frequency_setting current_freq =
```

Comment:
Isn't DVFS frequency always high right after boot (so that application must explicitly lower it when needed)?

Reply:
If we set it here, we would misalign with DVFS frequency that is actually used